Client/Frontend/Browser/Tabs/TabCell.swift

```diff
@@ -0,0 +1,318 @@
+            LegacyGridTabViewController.UX.cornerRadius + TabCellUILayout.borderWidth
+        }
+    }
+
+    private lazy var backgroundHolder: UIView = .build { view in
+        view.layer.cornerRadius =
+        LegacyGridTabViewController.UX.cornerRadius + TabCellUILayout.borderWidth
+        view.clipsToBounds = true
+    }
+
+    private lazy var faviconBG: UIView = .build { view in
+        view.layer.cornerRadius = HomepageViewModel.UX.generalCornerRadius
+        view.layer.borderWidth = HomepageViewModel.UX.generalBorderWidth
+        view.layer.shadowOffset = HomepageViewModel.UX.shadowOffset
+        view.layer.shadowRadius = HomepageViewModel.UX.shadowRadius
+    }
+
+    private lazy var screenshotView: UIImageView = .build { view in
+        view.contentMode = .scaleAspectFill
+        view.clipsToBounds = true
+        view.isUserInteractionEnabled = false
+    }
+
+    private lazy var titleText: UILabel = .build { label in
+        label.isUserInteractionEnabled = false
+        label.numberOfLines = 1
+        label.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .body, size: 12, weight: .semibold)
+    }
+
+    private lazy var closeButton: UIButton = .build { button in
+        button.setImage(UIImage.templateImageNamed(StandardImageIdentifiers.Large.cross), for: [])
+        button.imageView?.contentMode = .scaleAspectFit
+        button.contentMode = .center
+        button.imageEdgeInsets = UIEdgeInsets(equalInset: LegacyGridTabViewController.UX.closeButtonEdgeInset)
+    }
+
+    // MARK: UICollectionViewCell
+
+    override func prepareForReuse() {
+        // Reset any close animations.
+        super.prepareForReuse()
+        screenshotView.image = nil
+        backgroundHolder.transform = .identity
+        backgroundHolder.alpha = 1
+        self.titleText.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .body, size: 12, weight: .semibold)
+        layer.shadowOffset = .zero
+        layer.shadowPath = nil
+        layer.shadowOpacity = 0
+        isHidden = false
+    }
+
+    // MARK: - Auto Layout
+
+    private func setupConstraints() {
+        NSLayoutConstraint.activate([
+            backgroundHolder.topAnchor.constraint(equalTo: contentView.topAnchor),
+            backgroundHolder.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            backgroundHolder.bottomAnchor.constraint(equalTo: contentView.bottomAnchor),
+            backgroundHolder.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+
+            title.topAnchor.constraint(equalTo: backgroundHolder.topAnchor),
+            title.leftAnchor.constraint(equalTo: backgroundHolder.leftAnchor),
+            title.rightAnchor.constraint(equalTo: backgroundHolder.rightAnchor),
+            title.heightAnchor.constraint(equalToConstant: LegacyGridTabViewController.UX.textBoxHeight),
+
+            favicon.leadingAnchor.constraint(equalTo: title.leadingAnchor, constant: 6),
```

Comment:
This is copying over the existing code from `LegacyTabCell` and is mostly untouched. I can move all of these out into UX constants but if so it would probably make sense to clean up both classes if we think the legacy cell will still be in place for a while

Reply:
It's ok to fix only for the new class, it's a good idea to reuse the elements of the Legacy class and just tweak the details to match the new conventions

Reply:
Sounds good, will update